src/Select/index.tsx

```diff
@@ -0,0 +1,335 @@
+                    if (isHTMLOptionElement(child)) {
+                      return (
+                        <ListItemWrapper
+                          data-top-level-index={topLevelIndex}
+                          downshiftItems={items}
+                          element={child}
+                          getItemProps={getItemProps}
+                          key={
+                            child.props.value
+                              ? child.props.value.toString()
+                              : child.props.children
+                          }
+                        />
+                      );
+                    } else if (isHTMLOptgroupElement(child)) {
+                      return (
+                        <React.Fragment key={child.props.label}>
+                          {topLevelIndex > 0 && (
+                            <ListDivider data-top-level-index={topLevelIndex} />
+                          )}
+                          <ListHeading
+                            aria-label={child.props.label}
+                            role="group"
+                          >
+                            {child.props.label}
+                          </ListHeading>
+                          {React.Children.map(
+                            child.props.children as React.ReactElement<
+                              OptionProps,
+                              "option"
+                            >[],
+                            (optgroupChild) => {
+                              return (
+                                <ListItemWrapper
+                                  key={
+                                    optgroupChild.props.value
+                                      ? optgroupChild.props.value.toString()
+                                      : optgroupChild.props.children
+                                  }
+                                  downshiftItems={items}
+                                  element={optgroupChild}
+                                  getItemProps={getItemProps}
+                                />
+                              );
+                            }
+                          )}
+                        </React.Fragment>
+                      );
+                    }
+
+                    return null;
+                  })}
+              </List>
+            }
+            placement={placement}
+            triggerEvents="manual"
+            matchTriggerWidth={matchTriggerWidth}
+            trigger={
+              <Button
+                {...props}
+                {...getToggleButtonProps({ disabled })}
+                className={cx(
+                  css({
+                    textAlign: "left",
+                    maxWidth: {
+                      auto: undefined,
+                      small: 71,
+                      medium: 110,
+                      large: 157,
+                      "extra large": 188,
+                    }[size],
```

Comment:
NBD
my brain took a second to parse what was being keyed on, might be nice to either move the dictionary or the final value into a variable